### ISSUE TITLE
Annoying compile warnings

### ISSUE THREAD

**Issue by ibell** [fmtlib/fmt#414]
When I compile a library calling cppformat, I get lots of warnings like these when I compile on OSX with Xcode:

```
../../externals/cppformat/fmt/format.h:1165:3: warning: unused typedef
      'Assert1168' [-Wunused-local-typedef]
  FMT_STATIC_ASSERT(False<T>::value,
  ^
../../externals/cppformat/fmt/format.h:1160:15: note: expanded from macro
      'FMT_STATIC_ASSERT'
  typedef int FMT_CONCAT_(Assert, __LINE__)[(cond) ? 1 : -1] FMT_UNUSED
              ^
../../externals/cppformat/fmt/format.h:1158:28: note: expanded from macro
      'FMT_CONCAT_'
# define FMT_CONCAT_(a, b) FMT_CONCAT(a, b)
                           ^
../../externals/cppformat/fmt/format.h:1142:26: note: expanded from macro
      'FMT_CONCAT'
#define FMT_CONCAT(a, b) a##b
                         ^
<scratch space>:116:1: note: expanded from here
Assert1168
```

It would be awesome if you would kill these warnings somehow.

**Comment by foonathan** [comment]
The `FMT_UNUSED` at the should silence the warning.

Could you go to line 1180 of `format.h` and insert an `#error "foo"` in the else of that `#ifdef`?
And if that error triggers, could you provide the value of `FMT_GCC_VERSION`?
I can't compile it on XCode.

**Comment by vitaut** [comment]
I was able to reproduce it. The problem was in the too conservative GCC version check. The unused attribute is available at least in gcc 3.2.1: https://gcc.gnu.org/onlinedocs/gcc-3.2.1/gcc/Type-Attributes.html#Type%20Attributes. Fixed in https://github.com/fmtlib/fmt/commit/2fa4655af62d672954271ded4a2e5ce5319e5946. Thanks for reporting!

**Comment by ibell** [comment]
Great, thanks for fixing this issue.  Back to zero warnings in my library
